daemon/src/taker_cfd.rs

```diff
@@ -166,13 +171,25 @@ impl Actor {
             .send(load_all_cfds(&mut conn).await?)?;
         let cfd = load_cfd_by_order_id(order_id, &mut conn).await?;
 
+        let latest_announcement = self
+            .latest_announcement
+            .to_owned()
+            .context("Unaware of oracle's latest announcement.")?;
+        let nonce_pks = latest_announcement.nonce_pks;
```

Comment:
@da-kami You will have to change this to look up the correct announcement based on the ID included in the offer.